### PR TITLE
#2 Detect if comment or not

### LINKED ISSUE #2: Detect if comment or not
- [x] 2. Write a program to identify whether a given line is a comment or not.

### PR DESCRIPTION
**Pull request by shreyanshanchlia**
Write a program to identify whether a given line is a comment or not.

### CODE REVIEW

File: is comment or not/comment or not_0.cpp [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+//Remove comments from string s
+void RemoveComments(string _s)
+{
+    //split s into code and comments
+    string code;
+    string comment;
+
+    //determines prev char to determine next char
+    enum CommentType
```
Comment (Kishorsarswat) [inline is comment or not/comment or not_0.cpp:14]: excellent approach

File: is comment or not/comment or not_0.cpp [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            comment += *ch;
+        }
+        else // if (Current == CommentType::MultiLine)
+        {
+            //encountered * could be the end of multi-line comment if / follows
+            if(*ch == '*')
+            {
+                comment += *ch;
+                
+                ch++;
```
Comment (Kishorsarswat) [inline is comment or not/comment or not_0.cpp]: This right here breaks the flow. everything checked char by char but here it is incrementing ch. 
you could have added an enum like "multilineEnd" or something to coverup.

**Review by Kishorsarswat — CHANGES_REQUESTED** [review]

File: is comment or not/comment or not_0.cpp (hunk 2 above)
Comment (Kishorsarswat) [reply is comment or not/comment or not_0.cpp]: As i suspeected, it is resulting as a bug. you can not close a multiline comment with **/

Comment (shreyanshanchlia) [reply is comment or not/comment or not_0.cpp]: Fixed

**Review by Kishorsarswat — APPROVED** [review]
fixes are done!

**Pull request merged**